### ISSUE TITLE
Lag in VisualStudio on editing step lines within a spec.

### ISSUE THREAD

**Issue by ocary** [getgauge/gauge-visualstudio#225]
This is the follow up to defect 223.  https://github.com/getgauge/gauge-visualstudio/issues/223

### Expected behavior
Step edits occur in a reasonable amount of time; reasonable is very quick - nearly instantaneous for typing text and within a second or two for auto-complete suggestions to come up.

### Actual behavior
Within a spec file, editing step lines - either adding new lines or editing existing steps - there is a 25 to 30 second lag before the character and the autocomplete list shows up.

On typing a new character, the api.log file generates ~35MB of log files and the content is many lines like:
13:56:06.880 Api Request Received: messageType:GetStepValueRequest messageId:63657237366880 stepValueRequest:<stepText:"_[Step Text]_" > 
13:56:06.880 Sending API response: messageType:GetStepValueResponse messageId:63657237366880 stepValueResponse:<stepValue:<stepValue:"_[Different Step Text]_" parameterizedStepValue:"_[Different Step Text]_" > > 

### Steps to reproduce
Publicly available reproduction steps not known yet; I'm working on generating an example   ~1000 spec files, plus 50+ concepts.    ~400 steps across ~15 .cs files.

Screenshots attached are show a simplified use case, and a dotTrace overview of that case.  I cannot attach the dotTrace log file as it is 35MB even when zipped (gitHub limit is 10MB)

### Version Information

VisualStudio Version and Edition  :-  Professional 2015.  Update 3.   version 14.0.025431.01
Gauge VisualStudio Plugin Version :- nightly-2018-03-16

```
Gauge version: 0.9.7
Commit Hash: bc9784c

Plugins
-------
csharp (0.10.1)
flash (0.0.1)
html-report (4.0.3)
java (0.6.6)
screenshot (0.0.1)
spectacle (0.1.2)
xml-report (0.2.0)
```

![visualstudiolag](https://user-images.githubusercontent.com/6934721/37732336-bd013376-2d12-11e8-8fbe-8db7de6f0eb9.png)

![trace](https://user-images.githubusercontent.com/6934721/37733282-9521ca5c-2d15-11e8-8c11-2f4d04588eea.png)

![tracingdetails](https://user-images.githubusercontent.com/6934721/37733430-039c581c-2d16-11e8-831c-afea88da289e.png)

**Comment by sriv** [comment]
Should be available in nightly >= 4/4/18

**Comment by ocary** [comment]
@sriv - Yesterday I saw the 2018-04-04 nightly was there and have been using it since.   Lag when editing a step line within a spec is gone, and intellisense is showing up as expected.   Thank you - this is a big improvement in working with specs and very appreciated!


One issues seen on another persons machine is that the IntelliSense list got messed up, but we have not been able to reproduce.  Example step definition would be "click the button" but the intellisense list was showing "cliclick the button"  (the first few characters were repeated).  This was fixed by closing visual studio and re-opening, and we have not been able to reproduce.   If we do, we'll open a new issue.
